src/pages/AppDetail.svelte

```diff
@@ -1,17 +1,18 @@
 <script>
+  import { onMount } from "svelte";
+  import { fetchJSON } from "../state/api";
+
   export let params;
   const URL = `data/${params.app}/index.json`;
   let app;
-  fetch(URL)
-    .then((r) => r.json())
-    .then((ret) => {
-      app = ret;
-    });
-</script>
 
-<h1>{params.app}</h1>
+  onMount(async () => {
+    app = await fetchJSON(URL);
+  });
+</script>
 
 {#if app}
+  <h1>{params.app}</h1>
```

Comment:
If the app failed to load, the page would be blank. There's no real reason to put this inside of the conditional, the `params` object doesn't depend on the state of `fetchJSON`.